src/term.rs

```diff
@@ -0,0 +1,57 @@
+use pyo3::basic::PyObjectProtocol;
+use pyo3::prelude::*;
+use pyo3::types::PyType;
+use tantivy::schema;
+use crate::facet::Facet;
+use crate::field::Field;
+
+
+/// Term represents the value that the token can take.
+///
+/// This term can be used to access index and operate on that value
+/// for example index.delete_term(term) to delete all elements with that value
+#[pyclass]
+pub(crate) struct Term {
+    pub(crate) inner: schema::Term,
+}
+
+#[pymethods]
+impl Term {
```

Comment:
In the spirit of my changes, could we keep remove `Term` and `Field` and use a field_name in place?

Instead, we could have `.delete_term(field_name: &str, field_value: dynamicpythontype)` and do the resolution of the field name as well as the value to the internal schema defined type happen transparently when `.delete_term` is called.

Reply:
I pushed an update applying your criteria, I'm using Document `extract_value`to avoid repeating code, maybe a utils module would be usefull to store this utility functions.